### ISSUE TITLE
Cannot exclude Gtk4 windows from tiling

### ISSUE THREAD

**Issue by Nine-H** [pop-os/shell#954]
**(1) Issue/Bug Description:**

I can't get apps built against gtk4 excluded from tiling, there are a number of apps I use as widgets floating above my tiled windows.

**(2) Steps to reproduce (if you know):**

- Install the shortwave flatpak from <https://flathub.org/apps/details/de.haeckerfelix.Shortwave>
- Try to add it to your excluded windows
- when it fails press ctrl+shift+i and go to the global tab to verify it's running Gtk4.

**(3) Expected behavior:**

it floats over my windows.

**(4) Distribution (run `cat /etc/os-release`):**
 
Pop Os 20.04

**(5) Gnome Shell version:**

3.36

**(6) Pop Shell version (run `apt policy pop-shell` or provide the latest commit if building locally):**

```
pop-shell:
  Installed: 1.1.0~1619032486~20.04~969e2c5
  Candidate: 1.1.0~1619032486~20.04~969e2c5
  Version table:
 *** 1.1.0~1619032486~20.04~969e2c5 1001
       1001 http://ppa.launchpad.net/system76/pop/ubuntu focal/main amd64 Packages
       1001 http://ppa.launchpad.net/system76/pop/ubuntu focal/main i386 Packages
        100 /var/lib/dpkg/status
```

**(7) Where was Pop Shell installed from:**

Debian pkg? it's your os mate.

**(8) Monitor Setup (2 x 1080p, 4K, Primary(Horizontal), Secondary(Vertical), etc):**

1080p notebook display, but it doesn't work on my multihead desktop either.

**(9) Other Installed/Enabled Extensions:**

- just perfect
- dash to dock
- hide top bar
- and I disabled desktop icons because they're ugly

**(10) Other Notes:**

The app doesn't use the pop theme either but that's an entirely different issue.

**Comment by mmstick** [comment]
I don't think this is an issue with GTK4 windows specifically. The problem was caused by the WMCLASS parameter being set to an empty string, rather than a unique name for that application.

**Comment by Nine-H** [comment]
cheers, should I refer the issue to the app developer or is that an edge case the shell extension should take care of?
